src/main/java/org/primefaces/component/dataview/DataViewRenderer.java

```diff
@@ -235,21 +235,32 @@ protected void encodeGridLayout(FacesContext context, DataView dataview) throws
 
         if (grid != null) {
             ResponseWriter writer = context.getResponseWriter();
-
             int columns = grid.getColumns();
             int rowIndex = dataview.getFirst();
             int rows = dataview.getRows();
             int itemsToRender = rows != 0 ? rows : dataview.getRowCount();
             int numberOfRowsToRender = (itemsToRender + columns - 1) / columns;
-            String columnClass = DataView.GRID_LAYOUT_COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns);
+            String columnClass = DataView.GRID_LAYOUT_COLUMN_CLASS + " ";
+            if (ComponentUtils.isFlex(context, dataview)) {
```

Comment:
might want to store off the result of `boolean isFlex = ComponentUtils.isFlex(context, dataview)` to resuse

Reply:
done